hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -1382,6 +1391,8 @@ private void deleteInternal(Container container, boolean force)
         if (!success) {
           LOG.error("Failed to move container under " +
               hddsVolume.getDeleteServiceDirPath());
+          // Trigger a volume scan as rename failed
+          StorageVolumeUtil.onFailure(container.getContainerData().getVolume());
```

Comment:
Before this we also check `container.isEmpty()` and `logBlocksIfNonZero(container)` to make sure container is really empty or not, which will throw `IOException` as well if there is problem in volume as we are accessing disk and rocksdb here. So we need to do trigger volume scan at that place too. 
Also https://github.com/apache/ozone/pull/4655 which will refactor this method and will probably merge in a day, I think we should handle this change on top of that.

Reply:
Thanks @ashishkumar50 , updated the change.